### PR TITLE
Bump up to Scala 2.13.12; Bump up sbt; Fix errors and warnings

### PR DESCRIPTION
**Pull request by haoozi**
Bump up to Scala 2.13.12
Bump up SBT 
Fix errors and warnings
Tested with RocketChip, produces identical output with master branch (same checksum)

### CODE REVIEW

File: src/main/scala/Harness.scala [hunk 1]
```diff
@@ -4,9 +4,10 @@ import firrtl._
 import firrtl.ir._
 
 import java.io.Writer
+import scala.collection.mutable.ArrayBuffer
 
 object HarnessGenerator {
-  def harnessConnections(m: Module) = {
+  def ections(m: Module) = {
```
Comment (sbeamer) [inline src/main/scala/Harness.scala]: I can't tell if GitHub truncated the name in this view, but it appears to be now named `ections`. Is the intent of that clear?

**Review by sbeamer — CHANGES_REQUESTED** [review]
I will miss the procedure syntax and omitting parentheses (auto-application), but yes, both are deprecated for 2.13.

Just one minor name concern. Thanks for making this.

File: src/main/scala/Harness.scala (hunk 1 above)
Comment (haoozi) [reply src/main/scala/Harness.scala]: My bad, this looks like a mistake.

File: src/main/scala/Compiler.scala [hunk 2]
```diff
@@ -259,7 +259,7 @@ class EssentEmitter(initialOpt: OptFlags, w: Writer, circuit: Circuit) extends L
     if (initialOpt.writeHarness) {
       w.writeLines(0, "")
       w.writeLines(1, s"void connect_harness(CommWrapper<struct $topName> *comm) {")
-      w.writeLines(2, HarnessGenerator.harnessConnections(topModule))
+      w.writeLines(2, HarnessGenerator.ections(topModule))
```
Comment (haoozi) [inline src/main/scala/Compiler.scala]:
```suggestion
      w.writeLines(2, HarnessGenerator.harnessConnections(topModule))
```

File: src/main/scala/Harness.scala (hunk 1 above)
Comment (haoozi) [reply src/main/scala/Harness.scala:10]:
```suggestion
  def harnessConnections(m: Module) = {
```

**Review by haoozi — COMMENTED** [review]
`harnessConnect` is mistakenly renamed to `ections`. Revert back

File: src/main/scala/Harness.scala (hunk 1 above)
Comment (haoozi) [reply src/main/scala/Harness.scala]:
```suggestion
  def harnessConnections(m: Module) = {
```

**Review by sbeamer — APPROVED** [review]

**Comment by sbeamer** [comment]
Thanks!

**Pull request merged**